### DISCUSSION TITLE
[Unstable] New missing target SE warning error might ruin ton of mods

### DISCUSSION THREAD

**Discussion by PanmanS** [FakeFishGames/Barotrauma#13595] [Unstable feedback]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

A lot of mods rely on the default target for the status effect being `This`, with this new error, a ton of mods on workshop now throw errors. Not only that would prevent effective debugging because actual errors would covered under ton of those warnings,  in especially big mods with ton of items, like it was with my COL or like it is with barotraumatic, but it also apparently lowers your frames because otherwise I cannot explain why only on unstable my COL mod was perfoming so bad, until I added targets.

So please can we revert this change to how it is currently on unstable?

example: 
![Barotrauma_WQDFqiZrbm](https://github.com/FakeFishGames/Barotrauma/assets/42521948/891f45b0-6a9a-4da6-ba0f-dc3ff5ed737f)

### Reproduction steps

1. quickstart level or sub editor in stable version of game
2. spawn ton of items with status effects missing targets (so it default to `this`)
3. Record perfomance
4. Repeat same procedure on Unstable and observe massive perfomance drop or atleast cluttered console which wouldnt happen on stable.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.2.13.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

This error should actually only be logged when the status effect is being created, which shouldn't happen constantly, but e.g. in this case when the character first spawns. So I would suspect that the performance drop is actually caused by something else (unless the mod is actually spawning something constantly, which seems like a problem in itself).

I'm honestly a bit reluctant to revert this change, because forgetting to set the target on a StatusEffect has turned out to often cause issues, so it's better to warn modders (and us devs!) about these kinds of misconfigured effects and encourage them to explicitly define what the effect should target (e.g. whether it should be the limb or the entire character). It's better to fix issues than to sweep them under a rug! :)

**Comment by PanmanS** [comment]
Well about performance loss: I literally just put target="this" on 90% of SE that were missing em (a.k.a defaulting to `this`) and well perfomance increased. I think it happens because those SE appear and disappear multiple times. Once again on stable such perfomance issues were never experienced in a first place.


Quite sad to hearing that it will stay but ig it is for the better.

**Comment by SeaverBeaver** [comment]
Then why not make it a warning, like you did when the depth value system was changed?

A potential error that isn’t actually an error -because of the automatic correction to default- shouldn’t ~~spam open the console and make the mod unplayable.~~ throw a bunch of red messages in the console. (Does it open the console? If yes than it’s an error, if no than it’s a warning… right?)

Any elaboration on what the issues are when the StatusEffect target is not set?

**Comment by PanmanS** [comment]
Also recouloruing it to yellow instead of red would be nice

**Comment by Regalis11** [comment]
I think I'll need to look into this more closely, if it does indeed cause performance problems we definitely need to do something about it. And we could make it an actual orange warning instead, now it's this weird "normal" console message that's just colored red, but doesn't force the console to auto-open.

> Any elaboration on what the issues are when the StatusEffect target is not set?

The issue is that the game essentially has to guess what you meant: e.g. whether you intended it to target the limb it's defined in, or the character the limb belongs to. Usually it's fine to guess that the target is the limb, but not always: many vanilla monsters' limbs for example had OnDamaged effects with no target defined, which were intended to check certain conditions on the character and e.g. play sounds or emit particles, but they instead defaulted to targeting the limb and never executed. When the game asks you to define the target explicitly, it's much harder to do mistakes like this.

**Comment by ubertpendragon** [comment]
Came here to say it is indeed extremely annoying/distracting. 

I spent a quite some time seeking for statuseffects just to add target="this" just to make the error message go away
